builtin/game/chatcommands.lua

```diff
@@ -451,59 +457,130 @@ core.register_chatcommand("set", {
 	end,
 })
 
-local function emergeblocks_callback(pos, action, num_calls_remaining, ctx)
-	if ctx.total_blocks == 0 then
-		ctx.total_blocks   = num_calls_remaining + 1
-		ctx.current_blocks = 0
+local function emerge_callback(pos, action, num_calls_remaining, ctx)
+	if ctx.total_count == 0 then
+		ctx.total_count   = num_calls_remaining + 1
+		ctx.current_count = 0
 	end
-	ctx.current_blocks = ctx.current_blocks + 1
-
-	if ctx.current_blocks == ctx.total_blocks then
+	ctx.current_count = ctx.current_count + 1
+	if core.get_us_time() < ctx.cur_time then
+		-- core.get_us_time() loops after 2^32 microseconds.
+		-- The possibility of two loops having occurred (at least
+		-- 72 minutes have elapsed) is ignored deliberately as
+		-- it's unlikely, and non-critical anyway.
+		ctx.start_time = ctx.start_time - 2^32
```

Comment:
Lua 5.1 and LuaJit both do constant folding, although Lua 5.1 might miss a few opportunities e.g. when going against operator precedence.

LuaJit might even recognize constant upvalues and should fold over immutable variables then.

If you want to optimize, then removing global and table lookups or recycling tables  would have bigger effects. But is emerging called often enough for that to matter?